### PR TITLE
build: switch to ESRP v5, which supports managed identities

### PR DESCRIPTION
**Pull request by DHowett**
This required me to push a bunch more parameters through the build pipeline, but it gave me the opportunity to define them as variables that can be set at queue time.

### CODE REVIEW

**Comment by DHowett** [comment]
esrpv5 is, predictably, broke

**Comment by DHowett** [comment]
image

**Comment by DHowett** [comment]
image

File: build/pipelines/templates-v2/job-build-package-wpf.yml [hunk 1]
```diff
@@ -27,6 +27,9 @@ parameters:
   - name: publishArtifacts
     type: boolean
     default: true
+  - name: signingIdentity
```
Comment (zadjii-msft) [inline build/pipelines/templates-v2/job-build-package-wpf.yml:30]: heck at this rate, we should almost have a `do-codesign.yml` template that takes one of these `signingIdentity` objects we're passing around

**Review by zadjii-msft — APPROVED** [review]

File: build/pipelines/templates-v2/job-package-conpty.yml [hunk 2]
```diff
@@ -82,10 +85,15 @@ jobs:
...
+    - task: EsrpCodeSigning@5
       displayName: Submit *.nupkg to ESRP for code signing
       inputs:
-        ConnectedServiceName: 9d6d2960-0793-4d59-943e-78dcb434840a
+        ConnectedServiceName: ${{ parameters.signingIdentity.serviceName }}
+        AppRegistrationClientId: ${{ parameters.signingIdentity.appId }}
+        AppRegistrationTenantId: ${{ parameters.signingIdentity.tenantId }}
+        AuthAKVName: ${{ parameters.signingIdentity.akvName }}
+        AuthCertName: ${{ parameters.signingIdentity.authCertName }}
+        AuthSignCertName: ${{ parameters.signingIdentity.signCertName }}
```
Comment (lhecker) [inline build/pipelines/templates-v2/job-package-conpty.yml:96]: After googling a bit, I wonder if you can do
```yml
inputs:
  ${{ each var in parameters.signingIdentity }}:
    ${{var.name}}: ${{ var.value }}
  FolderPath: $(Build.ArtifactStagingDirectory)/nupkg
```

**Review by lhecker — APPROVED** [review]

File: build/pipelines/templates-v2/job-package-conpty.yml (hunk 2 above)
Comment (DHowett) [reply build/pipelines/templates-v2/job-package-conpty.yml:96]: Probably, yeah. I'm going to reach for something more like Mike's recommendation though - and maybe do both of these at the same time.

Comment (zadjii-msft) [reply build/pipelines/templates-v2/job-package-conpty.yml:96]: wait that sounds like a way to leak secrets. Like, if we add params, forget that we did this, and now we're passing all sorts of stuff to esrp that it wasn't expecting

**Pull request merged**